shared_models/shared_models.py

```diff
@@ -100,13 +103,54 @@ class Endorsement(BaseModel):
     state: str
 
 
+class ServiceDecorator(TypedDict):
+    endpoint: str
+    recipient_keys: List[str]
+    routing_keys: Optional[List[str]]
+
+
+class Oob(BaseModel):
+    role: Optional[str] = str
+    invi_msg_id: Optional[str] = None
+    state: Optional[str] = str
+    trace: Optional[str] = None
+    our_recipient_key: Optional[str] = None
+    oob_id: Optional[str] = None
+    invitation: Union[InvitationMessage, Mapping[str, Any]] = None
```

Comment:
I think this should be `InvitationMessage`. If the type isn't `InvitationMessage` when transforming, we should transform it to `InvitationMessage` (Otherwise we let the user deal with the incosistency instead of handling it for them)

Reply:
Same for `their_service` and `our_service`.